axonserver/src/main/java/io/axoniq/axonserver/config/AxonServerDiskMetricsConfig.java

```diff
@@ -0,0 +1,54 @@
+package io.axoniq.axonserver.config;
+
+import io.axoniq.axonserver.localstorage.LocalEventStore;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.binder.jvm.DiskSpaceMetrics;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+
+import java.nio.file.FileSystem;
+
+/**
+ * Configuration of Axon Server Disk metrics.
+ *
+ * @author Stefan Dragisic
+ * @since 4.5
+ */
+@Configuration
+@DependsOn("eventStoreManager")
+public class AxonServerDiskMetricsConfig {
+
+    private final Logger logger = LoggerFactory.getLogger(AxonServerDiskMetricsConfig.class);
+
+    private final LocalEventStore localEventStore;
+
+    public AxonServerDiskMetricsConfig(LocalEventStore localEventStore) {
```

Comment:
Disk space metrics may not be relevant for all types of event stores, The metrics are only relevant for the current implementation of the EventStorageEngine.